### PR TITLE
Addressed refactoring related review comments on PR #1775

### PR DESCRIPTION
**Pull request by priyanshi-yb**
Jenkins Schema pipeline - 
https://jenkins.dev.yugabyte.com/job/users/job/yb-voyager-testing/job/yb-voyager-testing-pipeline/3791/

Migration caveats
Screenshot 2024-10-30 at 8 47 03 PM

### CODE REVIEW

File: yb-voyager/cmd/templates/assessmentReport.template [hunk 1]
```diff
@@ -263,38 +263,47 @@
...
-                                {{range .Objects}}
-                                    <li class='list_item'>{{.ObjectName}}</li>
-                                {{end}}
-                            </ul>
-                        </div>
-                    {{end}}
-                    {{if .DocsLink}}
-                        <p id="refer-p"><a href="{{.DocsLink}}" target="_blank">Details</a></p>
-                    {{end}} 
+        {{ $hasMigrationCaveats := false }}
```
Comment (sanyamsinghal) [inline yb-voyager/cmd/templates/assessmentReport.template:271]: indentation

File: yb-voyager/cmd/templates/assessmentReport.template [hunk 2]
```diff
@@ -263,38 +263,47 @@
...
+        <h2>Migration caveats</h2>
+        {{range .MigrationCaveats}}
+            {{if .Objects}} <!-- Check if Objects is not empty -->
+                {{ $hasMigrationCaveats = true }}
+                {{if .DisplayDDL }} <!-- for these feature we are displaying the DDLs-->
```
Comment (sanyamsinghal) [inline yb-voyager/cmd/templates/assessmentReport.template:276]: please attach report screenshot of report for scenarios covered here.

**Review by sanyamsinghal — COMMENTED** [review]
Thanks for taking this up.
Overall looks good, few minor comments only.

File: yb-voyager/cmd/templates/assessmentReport.template (hunk 2 above)
Comment (priyanshi-yb) [reply yb-voyager/cmd/templates/assessmentReport.template:276]: Added.

**Pull request merged**